aspnetcore/fundamentals/dependency-injection.md

```diff
@@ -150,36 +150,78 @@ public void ConfigureServices(IServiceCollection services)
 }
 ```
 
-For more information, see the [ServiceCollection Class](/dotnet/api/microsoft.extensions.dependencyinjection.servicecollection) in the API documentation.
+For more information, see the <xref:Microsoft.Extensions.DependencyInjection.ServiceCollection> clss in the API documentation.
 
 ## Service lifetimes
 
 Choose an appropriate lifetime for each registered service. ASP.NET Core services can be configured with the following lifetimes:
 
-**Transient**
+### Transient
 
-Transient lifetime services are created each time they're requested from the service container. This lifetime works best for lightweight, stateless services.
+Transient lifetime services (<xref:Microsoft.Extensions.DependencyInjection.ServiceCollectionServiceExtensions.AddTransient*>) are created each time they're requested from the service container. This lifetime works best for lightweight, stateless services.
 
-**Scoped**
+### Scoped
 
-Scoped lifetime services are created once per client request (connection).
+Scoped lifetime services (<xref:Microsoft.Extensions.DependencyInjection.ServiceCollectionServiceExtensions.AddScoped*>) are created once per client request (connection).
 
 > [!WARNING]
 > When using a scoped service in a middleware, inject the service into the `Invoke` or `InvokeAsync` method. Don't inject via constructor injection because it forces the service to behave like a singleton. For more information, see <xref:fundamentals/middleware/index>.
 
-**Singleton**
+### Singleton
 
-Singleton lifetime services are created the first time they're requested (or when `ConfigureServices` is run and an instance is specified with the service registration). Every subsequent request uses the same instance. If the app requires singleton behavior, allowing the service container to manage the service's lifetime is recommended. Don't implement the singleton design pattern and provide user code to manage the object's lifetime in the class.
+Singleton lifetime services (<xref:Microsoft.AspNet.OData.Builder.ODataModelBuilder.AddSingleton*>) are created the first time they're requested (or when `Startup.ConfigureServices` is run and an instance is specified with the service registration). Every subsequent request uses the same instance. If the app requires singleton behavior, allowing the service container to manage the service's lifetime is recommended. Don't implement the singleton design pattern and provide user code to manage the object's lifetime in the class.
 
 > [!WARNING]
 > It's dangerous to resolve a scoped service from a singleton. It may cause the service to have incorrect state when processing subsequent requests.
 
+## Service registration methods
+
+Each service registration extension method offers overloads that are useful in specific scenarios.
+
+| Method | Automatic<br>object<br>disposal | Mocks for<br>testing | Pass args |
```

Comment:
```suggestion
| Method | Automatic<br>object<br>disposal | Alternate or multiple implementations | Pass args |
```

Maybe?

Or maybe just drop this column completely? The reasons on why you'd want to use an interface vs a concrete type are numerous and may be beyond the scope of this document (or this section at least). Mocking is one of many reasons.